lockrenewal_test.go

```diff
@@ -39,17 +63,18 @@ func (suite *serviceBusSuite) TestQueueSendReceiveWithLock() {
 	}
 }
 
-func testQueueSendAndReceiveWithRenewLock(ctx context.Context, t *testing.T, queue *Queue) {
+func testQueueSendAndReceiveWithRenewLock(ctx context.Context, t *testing.T, queue *Queue, numMessages int) {
 	ttl := 5 * time.Minute
-	numMessages := rand.Intn(40) + 20
 	activeMessages := make([]*Message, 0, numMessages)
 	expected := make(map[string]int, numMessages)
 	seen := make(map[string]int, numMessages)
 	errs := make(chan error, 1)
 
-	renewEvery := time.Second * 20
-	processingTime := time.Second * 100
-
+	renewEvery := time.Second * 6
+	// all are held in memory. we wait a bit longer than the lock expiry set to 10s
+	processingTime := 15 * time.Second
```

Comment:
All of this waiting stuff seems brittle to me.  Wouldn't it be better to use a `sync.WaitGroup` instead?  This is more general clean-up so I can take care of that at some future point.

Reply:
why is this brittle?
we receive all messages. to test auto-lock renewal you must wait that you have passed the peek lock timeout, called RenewLocks and then call Accept on the message. 
the lock times out after 10 seconds, the renew happens every 6 seconds, so I wait 15 to be sure all messages are received.
sure, you can make it more signal based... but I don't think that test will fail.
even if you do it signal based, you'll still want a context timeout in case the message aren't received, otherwise the test will hang.

Reply:
Well, it could be the case that I don't fully understand SB enough to convince myself it's not.  It's just when I see things like "wait a bit longer than..." it makes me think that if some other part of the test changes, then these timing might be off and the test starts failing.  Given it's been like this for quite some time it's likely fine, and if it does become a problem in the future we can address it then.

Reply:
yes I agree. but how else can you test a peek lock timeout that is controled serverside?